src/replica/replica_disk_migrator.cpp

```diff
@@ -148,12 +152,140 @@ bool replica_disk_migrator::check_migration_args(const replica_disk_migrate_requ
     return true;
 }
 
-// TODO(jiashuo1)
 // THREAD_POOL_REPLICATION_LONG
-void replica_disk_migrator::migrate_replica(const replica_disk_migrate_request &req) {}
+void replica_disk_migrator::do_disk_migrate_replica(const replica_disk_migrate_request &req)
+{
+    if (status() != disk_migration_status::MOVING) {
+        derror_replica("disk migration(origin={}, target={}), err = Invalid migration status({})",
+                       req.origin_disk,
+                       req.target_disk,
+                       enum_to_string(status()));
+        reset_status();
+        return;
+    }
+
+    if (init_target_dir(req) && migrate_replica_checkpoint(req) && migrate_replica_app_info(req)) {
```

Comment:
This entire execution spends too long, it will probably block other tasks on the REPLICATION_LONG thread pool.

Reply:
right, but the process need spend the long time, if not allow block `REPLICATION_LONG`, put `DEFAULT`?